samples/radio_test/src/radio_test.c

```diff
@@ -0,0 +1,565 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "radio_test.h"
+
+#include <string.h>
+#include <nrfx/hal/nrf_nvmc.h>
+#include <nrfx/hal/nrf_power.h>
+#include <nrfx/hal/nrf_rng.h>
+#include <nrfx_timer.h>
+#include <zephyr.h>
+
+/* IEEE 802.15.4 default frequency. */
+#define IEEE_DEFAULT_FREQ         (5)
+/* Length on air of the LENGTH field. */
+#define RADIO_LENGTH_LENGTH_FIELD (8UL)
+
+/* Frequency calculation for a given channel in the IEEE 802.15.4 radio
+ * mode.
+ */
+#define IEEE_FREQ_CALC(_channel) (IEEE_DEFAULT_FREQ + \
+				 (IEEE_DEFAULT_FREQ * \
+				 ((_channel) - IEEE_MIN_CHANNEL)))
+/* Frequency calculation for a given channel. */
+#define CHAN_TO_FREQ(_channel) (2400 + _channel)
+
+/* Buffer for the radio TX packet */
+static u8_t tx_packet[RADIO_MAX_PAYLOAD_LEN];
+/* Buffer for the radio RX packet. */
+static u8_t rx_packet[RADIO_MAX_PAYLOAD_LEN];
+/* Number of transmitted packets. */
+static u32_t tx_packet_cnt;
+/* Number of received packets with valid CRC. */
+static u32_t rx_packet_cnt;
+
+/**< Radio current channel (frequency). */
+static u8_t current_channel;
+
+/* Timer used for channel sweeps and tx with duty cycle. */
+static const nrfx_timer_t timer = NRFX_TIMER_INSTANCE(0);
+
+static u8_t rnd8(void)
+{
+	nrf_rng_event_clear(NRF_RNG, NRF_RNG_EVENT_VALRDY);
+
+	while (!nrf_rng_event_check(NRF_RNG, NRF_RNG_EVENT_VALRDY)) {
+		/* Do nothing. */
+	}
+	return nrf_rng_random_value_get(NRF_RNG);
+}
+
+static void radio_channel_set(nrf_radio_mode_t mode, u8_t channel)
+{
+#if USE_MORE_RADIO_MODES
+	if (mode == NRF_RADIO_MODE_IEEE802154_250KBIT) {
+		if ((channel >= IEEE_MIN_CHANNEL) &&
+			(channel <= IEEE_MAX_CHANNEL)) {
+			nrf_radio_frequency_set(
+				NRF_RADIO,
+				CHAN_TO_FREQ(IEEE_FREQ_CALC(channel)));
+		} else {
+			nrf_radio_frequency_set(
+				NRF_RADIO,
+				CHAN_TO_FREQ(IEEE_DEFAULT_FREQ));
+		}
+	} else {
+		nrf_radio_frequency_set(NRF_RADIO, CHAN_TO_FREQ(channel));
+	}
+#else
+	nrf_radio_frequency_set(NRF_RADIO, CHAN_TO_FREQ(channel));
+#endif /* USE_MORE_RADIO_MODES */
+}
+
+static void radio_config(nrf_radio_mode_t mode, enum transmit_pattern pattern)
+{
+	nrf_radio_packet_conf_t packet_conf;
+
+	/* Reset Radio ramp-up time. */
+	nrf_radio_modecnf0_set(NRF_RADIO, false, RADIO_MODECNF0_DTX_Center);
+	nrf_radio_crc_configure(NRF_RADIO, RADIO_CRCCNF_LEN_Disabled,
+				NRF_RADIO_CRC_ADDR_INCLUDE, 0);
+
+	/* Set the device address 0 to use when transmitting. */
+	nrf_radio_txaddress_set(NRF_RADIO, 0);
+	/* Enable the device address 0 to use to select which addresses to
+	 * receive
+	 */
+	nrf_radio_rxaddresses_set(NRF_RADIO, 1);
+
+	/* Set the address according to the transmission pattern. */
+	switch (pattern) {
+	case TRANSMIT_PATTERN_RANDOM:
+		nrf_radio_prefix0_set(NRF_RADIO, 0xAB);
```

Comment:
Just finished testing the PR on nRF52840, works like a charm!
I would recommend a bit more random number on the address ;-) how about PREFIX=0x6a. Base = 0x58fe811b.
Freshly generated from /dev/urandom.

Reply:
the new address is already in place